### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #397

Documentation improvements. No code changes or new tests in this PR, just documentation.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
OK, I have discovered that if we say "@ref moda_tababd" instead of just "tababd", then doxygen generates a nice link to that module.

So for example:

```
C> This subroutine initializes the internal arrays
C> (in module @ref moda_tababd) holding the DX BUFR table. It then

```
The second line generates a link to the documentation for moda_tababd, which is very helpful.

I have made this change for most or all of the existing references to modules in the documentation, but if you see one that is not referenced, or add new documentation referencing modules, then please use this notation to generate the hyperlink in the doxygen output.

File: src/elemdx.f
Comment (edwardhartnett) [inline src/elemdx.f:1]: In this file I applied emacs autoindent and used other automatic emacs tools to format the comments for line length and whitespace. Only formatting changes for this file.

File: src/modules_arrs.F90
Comment (edwardhartnett) [inline src/modules_arrs.F90:1]: In this file and in modules_vars.F90 I applied emacs autoindent and used other automatic emacs tools to format the comments for line length and whitespace. Only formatting changes for these files.

Comment (jbathegit) [reply src/modules_arrs.F90:1]: OK, but I thought one of the advantages of using F90 was that we could also have comments going out past column 72, and not just the source code going out past column 72.

Comment (edwardhartnett) [reply src/modules_arrs.F90:1]: Sure but we don't have to have extra long lines. ;-)

**Comment by jbathegit** [comment]
The @ref feature is a nice touch for the modules, so thanks for figuring that out!

FWIW, I'd only been using @ref previously to reference other points in the documentation that had already been tagged using @anchor.  So this is good to know that it works automatically for defined modules!

**Review by jbathegit — APPROVED** [review]

**Pull request merged**